java/org/puredata/core/utils/IoUtils.java

```diff
@@ -82,6 +82,14 @@ public static File extractResource(InputStream in, String filename, File directo
     directory.mkdirs();
```

Comment:
```suggestion
    directory.mkdirs();
    File canonicalRoot = directory.getCanonicalPath();
```

Reply:
Good idea. One correction: `getCanonicalPath()` returns `String`.

```suggestion
    directory.mkdirs();
    String canonicalRoot = directory.getCanonicalPath();
```